colossus/src/main/scala/colossus/core/WorkerManager.scala

```diff
@@ -1,84 +1,84 @@
 package colossus.core
 
-import akka.actor._
 import akka.pattern.ask
 import akka.routing.RoundRobinGroup
 import akka.util.Timeout
 import java.net.InetSocketAddress
 
+import akka.actor.{Actor, ActorRef, OneForOneStrategy, PoisonPill, Props, Stash, Terminated}
+import akka.actor.SupervisorStrategy.Restart
 import colossus.metrics.logging.ColossusLogging
 import colossus.{IOCommand, IOSystem}
 
 import scala.concurrent.Future
 import scala.concurrent.duration._
 import scala.collection.immutable.Iterable
+import scala.collection.mutable.ArrayBuffer
 import scala.util.{Failure, Success}
 
 /**
-  * A WorkerManager is just that, an Actor who is responsible for managing all of the Worker Actors in an IOSystem.
+  * A WorkerManager is an Actor that is responsible for managing all of the Worker Actors in an IOSystem.
   * It is responsible for creating, killing, restarting, relaying messages, etc.
   *
-  * @param workerAgent WorkerRefs that this WorkerManager manages
+  * @param workerRefs WorkerRefs that this WorkerManager manages
   * @param ioSystem Containing IOSystem
+  * @param workerFactory Worker factory for creating workers
   */
-private[colossus] class WorkerManager(workerAgent: IOSystem.WorkerAgent,
-                                      ioSystem: IOSystem,
-                                      workerFactory: WorkerFactory)
+private[colossus] class WorkerManager(workerRefs: IOSystem.WorkerRefs, ioSystem: IOSystem, workerFactory: WorkerFactory)
     extends Actor
     with ColossusLogging
     with Stash {
+
   import WorkerManager._
-  import akka.actor.OneForOneStrategy
-  import akka.actor.SupervisorStrategy._
-  import context.dispatcher
 
-  import ioSystem.numWorkers
+  import context.dispatcher
 
   /*
    * This strategy is chosen specifically for durability.  We don't want to kill the entire system if workers start
    * excepting.
    */
-  override val supervisorStrategy = {
+  override val supervisorStrategy: OneForOneStrategy = {
     OneForOneStrategy() {
-      case _: Exception => Restart
+      case e: Exception =>
```

Comment:
I don't think we need this supervisorStrategy since by default an exception was cause a restart. However, just to be extra cautious I will log first and see if this is actually happening in the real world.